debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -2088,6 +2088,100 @@ public void shouldStreamEnumAsKnownType() throws Exception {
+    // .with("column.propagate.source.type", "public.enum_array_table.value")
+    // .with(PostgresConnectorConfig.TABLE_WHITELIST, "public.enum_array_table"), false);
+
+    // waitForStreamingToStart();
+
+    // // We create the enum type after streaming started to simulate some future schema change
+
+    // TestHelper.execute("CREATE TYPE test_type AS ENUM ('V1','V2');");
+    // TestHelper.execute("ALTER TABLE enum_array_table ADD COLUMN value test_type[] NOT NULL");
+
+    // consumer = testConsumer(1);
+    // executeAndWait("INSERT INTO enum_array_table (value) VALUES ('{V1, V2}');");
+
+    // SourceRecord rec = assertRecordInserted("public.enum_array_table", PK_FIELD, 1);
+    // assertSourceInfo(rec, "postgres", "public", "enum_array_table");
+
+    // List<SchemaAndValueField> expected = Arrays.asList(
+    // new SchemaAndValueField(PK_FIELD, Schema.INT32_SCHEMA, 1),
+    // new SchemaAndValueField("value", SchemaBuilder.array(Enum.builder("V1,V2"))
+    // .parameter(TestHelper.TYPE_NAME_PARAMETER_KEY, "TEST_TYPE")
+    // .parameter(TestHelper.TYPE_LENGTH_PARAMETER_KEY, String.valueOf(Integer.MAX_VALUE))
+    // .parameter(TestHelper.TYPE_SCALE_PARAMETER_KEY, "0")
+    // .build(), Arrays.asList("V1", "V2")));
+
+    // assertRecordSchemaAndValues(expected, rec, Envelope.FieldName.AFTER);
+    // assertThat(consumer.isEmpty()).isTrue();
+
+    // // TODO: UPDATE
+    // // TODO: DELETE
+    // }
+
+    @Test
+    public void shouldStreamEnumArrayAsKnownType() throws Exception {
+        // Specifically enable `column.propagate.source.type` here to validate later that the actual
+        // type, length, and scale values are resolved correctly when paired with Enum types.
+        TestHelper.execute("CREATE TYPE test_type AS ENUM ('V1','V2');");
+        TestHelper.execute("CREATE TABLE enum_array_table (pk SERIAL, value test_type[] NOT NULL, PRIMARY KEY (pk));");
+        startConnector(config -> config
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NEVER)
+                .with("column.propagate.source.type", "public.enum_array_table.value")
+                .with(PostgresConnectorConfig.TABLE_WHITELIST, "public.enum_array_table"), false);
+
+        waitForStreamingToStart();
+
+        consumer = testConsumer(1);
+
+        // INSERT
+        executeAndWait("INSERT INTO enum_array_table (value) VALUES ('{V1, V2}');");
+
+        SourceRecord insertRec = assertRecordInserted("public.enum_array_table", PK_FIELD, 1);
+        assertSourceInfo(insertRec, "postgres", "public", "enum_array_table");
+
+        List<SchemaAndValueField> expectedInsert = Arrays.asList(
+                new SchemaAndValueField(PK_FIELD, Schema.INT32_SCHEMA, 1),
+                new SchemaAndValueField("value", SchemaBuilder.array(Enum.builder("V1,V2"))
+                        .parameter(TestHelper.TYPE_NAME_PARAMETER_KEY, "_TEST_TYPE")
```

Comment:
I want to call out that I had to change this from `TEST_TYPE` to `_TEST_TYPE`. Is that expected?